### ISSUE TITLE
Repeater enable session tracking unexpected behavior.

### ISSUE THREAD

**Issue by CookieAndPizza** [zaproxy/zaproxy#3888]
**To clarify: I do not know if this is the intended behavior or that this is a bug, but the behavior seems odd to me.**

**OWASP ZAP used: Version 2.6.0.**

Steps to reproduce this problem:

1. Open a request/response in the resender. The Set-Cookie header should be in the response
2. Make sure the "Enable session tracking" is disabled (Found under Edit -> Enable session tracking)
3. Send the just opened request

**Observed behavior:** 
The cookie (From the set cookie header) is now present in the request headers.

**Expected behavoir:**
Since session tracking is disabled, I would think that the request sent by me is unaltered. \

**Note:**
The same behavior happens when I enable session tracking

Thanks in advance!

**Comment by thc202** [comment]
That's the intended behaviour (since Paros days), but, yeah, from user's perspective it might be unexpected (I don't think it's documented anywhere that it will use its own HTTP state, if not using the global HTTP state).

**Comment by CookieAndPizza** [comment]
So if I understand correctly, there is no way to disable the automatic cookie placement in the request? I have tried (I think) every possible way of configuration. Since the application will always either use global, or the own http state.

**Comment by thc202** [comment]
Correct, there's a bug in the global HTTP state option that prevents that. There are a couple of issues related to the HTTP state and the request editor.

What's the best behaviour not have a HTTP state by default and just use the global HTTP State if/when enabled?

**Comment by thc202** [comment]
BTW, which version are you using to debug? 2.6.0? (With the latest develop/weekly it's possible to have the request editor not use any state at all with a workaround.)

**Comment by kingthorin** [comment]
> What's the best behaviour not have a HTTP state by default and just use the global HTTP State if/when enabled?

For my 2 cents it seems most user friendly/intuitive for the request editor to use the global HTTP state if/when enabled.

**Comment by CookieAndPizza** [comment]
I have been using ZAP 2.6.0 for normal use. When I'm debugging I have been using the development build pulled last week. 

About the workaround, do you refer to the following steps?
1. Enable global HTTP session tracking
2. Open the request editor
3. Select the cookie "Use current tracking session"
4. Disable global HTTP session tracking
5. Send request

In my attempts, this has been able to send the request in "ignore_cookie" mode (in the dev build). 

To answer your question,
I think the request editor should not use any http state (and thus use the ignore_cookie mode) unless the user enables the option "use global HTTP session tracking", then it should use the global state (therefore using the "browser_compatibility" option or the "default" option (I think it is defaulted to rfc_2109?)). 
I must say I do not fully understand the difference between the global http states and the non-global variant, could you elaborate on this? What is the functional use of a non-global http state tracking in the request editor?

**Comment by kingthorin** [comment]
I believe this is probably addressed (or at least in a much more userfriendly state) following: https://github.com/zaproxy/zaproxy/pull/4340
